Add Memgraph

kgolubic opened the pull request:
<!-- Congrats on creating an Awesome list! 🎉 -->

<!-- Please fill in the below placeholders -->

**(https://github.com/kgolubic/awesome-memgraph)**

**Memgraph is an in-memory graph database. I'm curating Memgraph Awesome list and I believe that it is a good add-on to this list.**

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

I've reviewed following PRs:

- [Add Scapy](https://github.com/sindresorhus/awesome/pull/2285)
- [Add ZkEVM](https://github.com/sindresorhus/awesome/pull/2278)
- [Add Computational Geometry](https://github.com/sindresorhus/awesome/pull/2275)

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions.
- You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The title of your entry should be title-cased and the URL to your list should end in `#readme`.
	- Example: `- [Software Architecture](https://github.com/simskij/awesome-software-architecture#readme) - The discipline of designing and building software.`
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it. **Instead use [#2242](https://github.com/sindresorhus/awesome/issues/2242) for incubation visibility**.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/main/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=main&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/main/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

kgolubic commented:
HI! I've fixed all of the errors reported by lint. I think that the remaining errors are false positives. I do have some links listed twice, but they appear in a different context. If this is not ok I can remove the duplicate links.

villesundell commented:
@kgolubic We have the same situation, and we opted to `lint disable double-link` as seen here: https://raw.githubusercontent.com/MystenLabs/awesome-move/446c2b511720c1c8f01af5d7f02861f3635121e7/README.md (the first line). Linting is part of Awesome's PR process, so in my understanding it must pass. (Awesome-move is not part of the "awesomeverse" yet.)

kgolubic commented:
unicorn

kgolubic commented:
I don't know how to handle one error "The repository should have a description". Can someone give me pointers on this one?

kgolubic commented:
> @kgolubic We have the same situation, and we opted to `lint disable double-link` as seen here: https://raw.githubusercontent.com/MystenLabs/awesome-move/446c2b511720c1c8f01af5d7f02861f3635121e7/README.md (the first line). Linting is part of Awesome's PR process, so in my understanding it must pass. (Awesome-move is not part of the "awesomeverse" yet.)

Thanks for the pointers. I've managed to fix all errors and `npx awesome-lint https://github.com/kgolubic/awesome-memgraph` gives me a passing mark.

What should I do now? Can I somehow re-run the check here on GitHub?

kgolubic commented:
@sindresorhus can you please tell me what should I do now? Can I somehow re-run the check here on GitHub? Local `lint `tells me that everything is ok now. Should I close this PR and open a new one?

0Lola commented:
Hi @kgolubic,

Here are some of the points I think you can improve on [readme](https://github.com/kgolubic/awesome-memgraph#readme):

* Remove `Inspired by the awesome list thing. Feel free to improve this list by contributing!`

> Doesn't include an Inspired by awesome-foo or Inspired by the Awesome project kinda link at the top of the readme. The Awesome badge is enough.

* Remove unknown format word `==============`.

* Add Memgraph Logo on the top-right like [this](https://github.com/sindresorhus/awesome-electron).
> Includes a project logo/illustration whenever possible.

* Contents only have one level, so `Resource/Documentation`, `Resource/Docker image`, `Resource/Tools` are wrong.
> Should only have one level of nested lists, preferably none.

* Fix format of the entries description.
> The link and description are separated by a dash.
Example: - [AVA](…) - JavaScript test runner.

❌ [MAGE 1.1 Live Demo - Analyzing a Twitter network with PageRank, node2vec & Community detection](https://www.youtube.com/watch?v=dn154j4sypo)

✅ [MAGE 1.1 Live Demo](https://www.youtube.com/watch?v=dn154j4sypo) - Analyzing a Twitter network with PageRank, node2vec & Community detection

Hope that helps. 👍

kgolubic commented:
All checks have passed ok.

sindresorhus commented on readme.md:
```diff
@@ -658,6 +658,7 @@
 - [TypeDB](https://github.com/vaticle/typedb-awesome#readme) - Logical database to organize large and complex networks of data as one body of knowledge.
 - [Cassandra](https://github.com/Anant/awesome-cassandra#readme) - Open-source, distributed, wide column store, NoSQL database management system.
 - [TDengine](https://github.com/taosdata/awesome-tdengine#readme) - An open-source time-series database with high-performance, scalability, and SQL support.
+- [Memgraph](https://github.com/kgolubic/awesome-memgraph#readme) - Streaming graph application platform, powered by an in-memory graph database built to handle real-time use-cases at enterprise scale.
```
Try to shorten the description.

kgolubic replied:
Done. I've made it shorter.

sindresorhus commented:
There are many things not followed in the guidelines.

You have also not addressed the feedback in https://github.com/sindresorhus/awesome/pull/2287#issuecomment-1221757581.

---

> Includes a succinct description of the project/theme at the top of the readme.

kgolubic commented:
> Hi @kgolubic,
> 
> Here are some of the points I think you can improve on [readme](https://github.com/kgolubic/awesome-memgraph#readme):
> 
> * Remove `Inspired by the awesome list thing. Feel free to improve this list by contributing!`
> 
> > Doesn't include an Inspired by awesome-foo or Inspired by the Awesome project kinda link at the top of the readme. The Awesome badge is enough.
> 
> * Remove unknown wording `==============`.
> * Add Memgraph Logo on the top-right like [this](https://github.com/sindresorhus/awesome-electron).
> 
> > Includes a project logo/illustration whenever possible.
> 
> * Fix format of the entries description.
>   ❌ [MAGE 1.1 Live Demo - Analyzing a Twitter network with PageRank, node2vec & Community detection](https://www.youtube.com/watch?v=dn154j4sypo)
>   ✅ [MAGE 1.1 Live Demo](https://www.youtube.com/watch?v=dn154j4sypo) - Analyzing a Twitter network with PageRank, node2vec & Community detection.
> 
> > The link and description are separated by a dash.
> > Example: - [AVA](%E2%80%A6) - JavaScript test runner.
> 
> Hope that helps.

I've updated the README.md file in accordance with your comments. Thank you!

kgolubic commented:
> There are many things not followed in the guidelines.
> 
> You have also not addressed the feedback in [#2287 (comment)](https://github.com/sindresorhus/awesome/pull/2287#issuecomment-1221757581).
> 
> > Includes a succinct description of the project/theme at the top of the readme.

Thank you for pointing this out. I've fixed all of the mentioned issues, added the logo and description on the top, shortened the description for te main Aweosnm list, and separated descriptions of links with "`-`". I hope that I've covered all of the issues.

kgolubic commented:
> There are many things not followed in the guidelines.
> 
> You have also not addressed the feedback in [#2287 (comment)](https://github.com/sindresorhus/awesome/pull/2287#issuecomment-1221757581).
> 
> > Includes a succinct description of the project/theme at the top of the readme.

@sindresorhus Thank you for pointing this out. I've fixed all of the mentioned issues, added the logo and description on the top, shortened the description for te main Aweosnm list, and separated descriptions of links with "-". I hope that I've covered all of the issues.

sindresorhus commented:
> The image should be high-DPI. Set it to maximum half the width of the original image.

---

Missing descriptions: https://github.com/kgolubic/awesome-memgraph#tools

---

> The description starts with an uppercase character and ends with a period.

---

"How Node2Vec Works - A Random Walk-Based Node Embedding Method"

Descriptions should not be title-cased.

kgolubic commented:
Thank you @sindresorhus and @Lissy93. I've updated my list according to your comments.

grem11n commented:
@kgolubic, according to this [awesome-lint rule](https://github.com/sindresorhus/awesome-lint/blob/main/rules/list-item.js#L185) descriptions in a list should end with a punctuation mark such as dot. There are a couple of occasions where your list is missing it. For example:

```markdon
### Tools

- [Memgraph Lab](https://memgraph.com/lab) - a visual user interface that helps you explore and manipulate the data stored in Memgraph
```

Other descriptions have this problem as well.

sindresorhus commented:
Bump

kgolubic commented:
> Bump

Sorry for the late answer. I missed the previous comment. I've updated descriptions with dots where they were needed.

sindresorhus commented:
> The description **starts with an uppercase character** and ends with a period.

kgolubic commented:
Ok. Thanks for all of the feedback. When I find the time, I'll improve the list an submit it again for consideration.
